src/compiler/emitter.ts

```diff
@@ -2288,7 +2288,8 @@ namespace ts {
                 writePunctuation(".");
             }
 
-            emitTokenWithComment(token.kind, node.expression.end, writePunctuation, node);
+            if (node.questionDotToken) writeTokenNode(node.questionDotToken, writePunctuation);
+            else emitTokenWithComment(token.kind, node.expression.end, writePunctuation, node);

```

Comment:
What happens if we just do this instead, since we already have the `token`?

```suggestion
            writeTokenNode(token, writePunctuation);
```

Reply:
That indeed simplifies it further...  (And I'm guessing that in the `textChanges.ts` context it makes the `onBeforeEmitToken`/`onAfterEmitToken` hooks run on the fake node too, which shouldn't do any harm.)  I pushed an update with this.